app/data/data-travel-certificate/data-travel-certificate-public/src/main/kotlin/com/hedvig/android/data/travelcertificate/GetEligibleContractsWithAddressUseCase.kt

```diff
@@ -0,0 +1,37 @@
+package com.hedvig.android.data.travelcertificate
+
+import arrow.core.Either
+import com.apollographql.apollo3.ApolloClient
+import com.apollographql.apollo3.cache.normalized.FetchPolicy
+import com.apollographql.apollo3.cache.normalized.fetchPolicy
+import com.hedvig.android.apollo.safeExecute
+import com.hedvig.android.apollo.toEither
+import com.hedvig.android.core.common.ErrorMessage
+import octopus.EligibleContractsWithAddressQuery
+
+interface GetEligibleContractsWithAddressUseCase {
+  suspend fun invoke(): Either<ErrorMessage, List<ContractEligibleWithAddress>>
+}
+
+internal class GetEligibleContractsWithAddressUseCaseImpl(
+  private val apolloClient: ApolloClient,
+) : GetEligibleContractsWithAddressUseCase {
+  override suspend fun invoke(): Either<ErrorMessage, List<ContractEligibleWithAddress>> {
+    return apolloClient.query(EligibleContractsWithAddressQuery())
+      .fetchPolicy(FetchPolicy.NetworkOnly)
+      .safeExecute()
+      .toEither(::ErrorMessage)
+      .map {
+        it.currentMember.activeContracts.filter { contract ->
+          contract.supportsTravelCertificate
+        }.map { contract ->
+          ContractEligibleWithAddress(contract.exposureDisplayName.substringBefore("•"), contract.id)
```

Comment:
I checked what iOS does to get the name for this. The check inside `travelCertificateSpecifications.contractSpecifications.location.street` and that one gives the right string without having to rely on there being a `•` to format it right, since that may not always really be true now or in the future.

Can you give this a shot. Change the query to
```
query CurrentMember {
	currentMember {
    activeContracts {
      id
      supportsTravelCertificate
    }
    travelCertificateSpecifications {
      contractSpecifications {
        contractId
        location {
          street
        }
      }
    }
	}
}
```
And then you gotta match the `activeContracts.id` with `contractId` from inside `contractSpecifications`, and get the `street` from inside there.

For reference, this is a sample response for this.
![image](https://github.com/HedvigInsurance/android/assets/44558292/666e1096-e5e5-4f2b-9962-eadcd390e961)

And one with the exposureName which shows the difference between the two
![image](https://github.com/HedvigInsurance/android/assets/44558292/76cda50e-d1df-436e-9253-05837544fbc2)

Reply:
Yes, sure!
I actually had a thought that probably it makes sense to show the number of insured people on this address too? Since on this depends who you can add to the certificate on the next step. Maybe just leave the whole exposureDisplayName there without the substrings?